images/runtime/node/installDependencies.sh

```diff
@@ -11,36 +11,40 @@ set -ex
 mkdir -p /node_modules
 chmod 777 /node_modules
 
-# Npm version 6.4 has issues installing native modules like grpc,
-# so upgrading it to a version which we know works fine.
-npm_ver=$(npm --version)
-IFS='.' read -ra SPLIT_VERSION <<< "$npm_ver"
-major="${SPLIT_VERSION[0]}"
-minor="${SPLIT_VERSION[1]}"
-if [ "$major" == "6" ] && [ "$minor" == "4" ]; then
-    echo
-    echo "Upgrading npm version from $npm_ver to 6.9.0"
-    npm install -g npm@6.9.0
-fi
-
-# Do NOT install PM2 from Node 14 onwards
-# v10.14.0
+# Since older versions of npm cli have security vulnerabilities, we try upgrading it
+# to the latest available version. However latest versions of npm do not work with very
+# old versions of Node (for example: 4 or 6), so we special case here to limit the upgrade
+# to only versions 8 or above
 currentNodeVersion=$(node --version)
-echo "Current Node version is $currentNodeVersion"
+# v10.4.5 => 10.4.5
 currentNodeVersion=${currentNodeVersion#?}
-IFS='.' read -ra SPLIT_VERSION <<< "$currentNodeVersion"
-major="${SPLIT_VERSION[0]}"
+echo "Current Node version is $currentNodeVersion"
+IFS='.' read -ra NODE_VERSION_PARTS <<< "$currentNodeVersion"
+nodeVersionMajor="${NODE_VERSION_PARTS[0]}"
 
-if [ "$major" -lt "14" ]; then
+currentNpmVersion=$(npm --version)
+echo "Version of npm: $currentNpmVersion"
+
+# Upgrade npm to the latest available version
+if [[ $nodeVersionMajor -ge 10  ]]; then
+    echo "Upgrading npm..."
+    npm install npm -g -loglevel silent
+    echo "Done upgrading npm."
+    currentNpmVersion=$(npm --version)
+    echo "Version of npm after upgrade: $currentNpmVersion"
+fi
+
+# Do NOT install PM2 from Node 14 onwards
+if [[ $nodeVersionMajor -lt 14 ]]; then
     echo "Installing PM2..."
     # PM2 is supported as an option when running the app,
     # so we need to make sure it is available in our images.
-    npm install -g pm2@3.5.1
+    npm install -g pm2@3.5.1 -loglevel silent
 else
     echo "Skipping PM2 installation..."
 fi
 
 # Application-Insights is supported as an option for telemetry when running the app,
 # so we need to make sure it is available in our images.
 # Updated to 1.8.3 that doesn't emit json payload in stdout which is causing issues to customers in ant-88
-npm install -g applicationinsights@1.8.3
+npm install -g applicationinsights@1.8.3 -loglevel silent
```

Comment:
could you move the appinsights version and pm2 version as constants

Reply:
Sure, sounds good